_test/stackalize

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+
+shopt -s extglob
+
+exampleFile="example.hs"
+testSuffix="_test.hs"
+srcDir="src"
+testDir="test"
+programName=`basename "$0"`
+
+depsFinder="`dirname ${BASH_SOURCE}`/dependencies"
+
+generateStackYaml () {
+    echo "resolver: ${stackResolver}"
+}
+
+generatePackageYaml () {
+    local exercise=`basename ${PWD}`
+    local commonDeps=`${depsFinder} --examples-and-tests ${exercise}`
+    local exampleDeps=`${depsFinder} --only-in-examples ${exercise}`
+    local testDeps=`${depsFinder} --only-in-tests ${exercise}`
+
+    echo "name: $exercise"
+    echo ""
+    echo "dependencies:"
+
+    for dependency in ${commonDeps}; do
+      echo "  - ${dependency}"
+    done
+
+    echo ""
+    echo "library:"
+    echo "  source-dirs: ${srcDir}"
+    echo "  dependencies:"
+
+    for dependency in ${exampleDeps}; do
+      echo "    - ${dependency}"
+    done
+
+    echo ""
+    echo "tests:"
+    echo "  test:"
+    echo "    main: ${exercise}${testSuffix}"
+    echo "    source-dirs: ${testDir}"
+    echo "    dependencies:"
+    echo "      - ${exercise}"
+
+    for dependency in ${testDeps}; do
+      echo "      - ${dependency}"
+    done
+}
+
+stackalizeProject () {
+    local exercise=`basename ${PWD}`
+    local module=`grep "^module" ${exampleFile} | cut -d " " -f 2`
+    if [ ! -e "${exampleFile}"           ] ||
+       [ ! -e "${exercise}${testSuffix}" ] ; then
+       echo "$0: Cannot stackalize. Exercise files not found."
+       exit 1
+    else
+        mkdir .extraFiles                                    &&
+        mv * .extraFiles                                     &&
```

Comment:
First, you need to know that this is not the best example of *bash scripting* in the world. It's just something that I improvised to solve this problem. :smile: 

What the script does is:

- move everything to `.extraFiles`, so we get a known directory struture.
- move from there the *example* and *test* files to their correct locations.
- generate a `stack.yaml` with the provided resolver.
- generate a `package.yaml` (simpler than a *.cabal* file) with the correct dependencies.

It can centainly be improved, but It's been 17 years since the last time I played with `bash` as a sysadmin. Let's say it works under normal conditions. :grin: 

About your question, if there are additional source files in the directory, as in `forth` and `zipper`, they could be included accidentally in the project and break the build. Also, I wanted to be able to *unstackalize* the project, moving everything back to their places.